### PR TITLE
Update .gitignore from .hgignore

### PR DESCRIPTION
**Pull request by tkrodriguez**
I pulled over a few rules from .hgignore to ignore more things by default.  Some of the things in .hgignore are no longer required because we split off jvmci.  Anything else showing up for anyone?

cc @axel22 @gilles-duboscq @rschatz @thomaswue

### CODE REVIEW

**Comment by rschatz** [comment]
We could also add the `graal.options` file.

**Comment by gilles-duboscq** [comment]
regarding
* `*.iml`
* `*.class`
* `*.jar`
Do we need them? I prefer to ignored the exact places where these things should be so that we notice when these files appear where they shouldn't

**Comment by tkrodriguez** [comment]
We don't necessarily need them.  I view those as file kinds we'd never want to check into the repo so ignoring them everywhere is fine.  Would you like me to remove them?

**Comment by tkrodriguez** [comment]
I added /mx.graal-core/graal.options as well.

**Comment by tkrodriguez** [comment]
And *.bgv.

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+*.cfg
+*.bgv
+*.jar
+*.zip
+*.rej
+*~
+core.*
 /mx.graal-core/env
 /mx.graal-core/eclipse-launches/
+/mx.graal-core/graal.options
```
Comment (zapster) [inline .gitignore]: I think this should be `/graal.options`. (cf #83)

**Comment by gilles-duboscq** [comment]
> We don't necessarily need them. I view those as file kinds we'd never want to check into the repo so ignoring them everywhere is fine. Would you like me to remove them?
No strong opinion. Both have their virtues so i'm also fine with leaving them in.

File: .gitignore (hunk 1 above)
Comment (tkrodriguez) [inline .gitignore]: Thanks.  I was guessing from reading the mx.py and thought it would be in the same location as the env.  Fixed.

**Pull request merged**